### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'bind')

### ISSUE THREAD

**Issue by Remydeme** [Tirke/node-cache-manager-ioredis#4]
I get this error when I try to inject Redistore inside my CachingInterceptor. 

Code : 

```typescript
import { CacheModule, Module } from '@nestjs/common'
import RedisStore from '@tirke/node-cache-manager-ioredis'
import type { ClientOpts } from 'redis3'

import { AppConfig } from '@hublo/configuration/configs'

import type { CachedAppConfig } from './configuration'

@Module({
  imports: [
    CacheModule.registerAsync<ClientOpts>({
      isGlobal: true,
      inject: [AppConfig],
      useFactory: (config: CachedAppConfig) => {
        return config.useRedisStore
          ? {
              store: RedisStore,
              host: config.cache.redisHost,
              port: config.cache.redisPort,
              ttl: config.cache.ttl,
            }
          : {}
      },
    }),
  ],
})
```

Bug description : 

``` js 
TypeError: Cannot read properties of undefined (reading 'bind')
    at Object.caching (/Users/remydeme/WebstormProjects/hublo/monorepo/node_modules/.pnpm/cache-manager@4.1.0/node_modules/cache-manager/lib/caching.js:306:31)
    at InstanceWrapper.useFactory [as metatype] (/Users/remydeme/WebstormProjects/hublo/monorepo/node_modules/.pnpm/@nestjs+common@9.0.8_mgy7w2bm3yhmntrbe3boyroj6m/node_modules/@nestjs/common/cache/cache.providers.js:20:32)
    at Injector.instantiateClass (/Users/remydeme/WebstormProjects/hublo/monorepo/node_modules/.pnpm/@nestjs+core@9.0.8_t6wygpagnzylhnpg25zual64b4/node_modules/@nestjs/core/injector/injector.js:343:55)
    at callback (/Users/remydeme/WebstormProjects/hublo/monorepo/node_modules/.pnpm/@nestjs+core@9.0.8_t6wygpagnzylhnpg25zual64b4/node_modules/@nestjs/core/injector/injector.js:53:45)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at Injector.resolveConstructorParams (/Users/remydeme/WebstormProjects/hublo/monorepo/node_modules/.pnpm/@nestjs+core@9.0.8_t6wygpagnzylhnpg25zual64b4/node_modules/@nestjs/core/injector/injector.js:132:24)
    at Injector.loadInstance (/Users/remydeme/WebstormProjects/hublo/monorepo/node_modules/.pnpm/@nestjs+core@9.0.8_t6wygpagnzylhnpg25zual64b4/node_modules/@nestjs/core/injector/injector.js:57:13)
    at Injector.loadProvider (/Users/remydeme/WebstormProjects/hublo/monorepo/node_modules/.pnpm/@nestjs+core@9.0.8_t6wygpagnzylhnpg25zual64b4/node_modules/@nestjs/core/injector/injector.js:84:9)
    at async Promise.all (index 3)
    at InstanceLoader.createInstancesOfProviders (/Users/remydeme/WebstormProjects/hublo/monorepo/node_modules/.pnpm/@nestjs+core@9.0.8_t6wygpagnzylhnpg25zual64b4/node_modules/@nestjs/core/injector/instance-loader.js:47:9)
```

**Comment by Tirke** [comment]
Sorry @Remydeme that problem was emerging from the default export of the lib. I've switched to a named export for easier handling.
You can now use [`2.0.0`](https://github.com/Tirke/node-cache-manager-ioredis/releases/tag/%40tirke%2Fnode-cache-manager-ioredis%402.0.0) of the package who should fix your issue.

**Comment by curtdept** [comment]
@Tirke I'm hitting this using it in a commonjs legacy app we have.

**Comment by Tirke** [comment]
@curtdept Yep sorry for that, I wanted to provide more explicit, type-safe ways to init but there was also a bug in that the default constructor didn't use the params for the root `cache-manager` object. Everything should be fixed in `@tirke/node-cache-manager-ioredis@2.1.0`.
